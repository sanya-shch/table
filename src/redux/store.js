// @flow

import { createStore , applyMiddleware  } from "redux";
import reducers from "./reducers";
import type { Store as AppState } from "./reducers/tableReducer";
import { composeWithDevTools } from "redux-devtools-extension";
import type { Store } from "redux";
import type { ActionType } from "./actions/tableActions";

const store: Store<AppState, ActionType> = createStore(
  reducers,
  composeWithDevTools(applyMiddleware())
);

export default store;


