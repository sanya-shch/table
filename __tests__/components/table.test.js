import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Table from "../../src/components/Table";

import { Provider } from "react-redux";
import { createStore } from "redux";
import reducers from "../../src/redux/reducers/tableReducer";
import { testStore } from "../reducers/reducers.test";

describe("the Table component", () => {
  it("table", () => {
    const store = makeTestStore();

    const { getByText, getAllByText, container } = testRender(<Table />, {
      store
    });

    expect(container).toMatchSnapshot();

    expect(getAllByText("Delete").length).toEqual(3);

    expect(getByText("Add").className).toEqual("addBtn");

    fireEvent.click(getByText("Add"));

    expect(store.dispatch).toBeCalled();

    expect(store.dispatch).toMatchSnapshot();
    expect(store.dispatch).toHaveBeenCalledWith({ type: "ADD_ROW" }); //.toBe();

    //

    fireEvent.click(getAllByText("Delete")[0]);
    expect(store.dispatch).toHaveBeenCalledWith({
      payload: "0",
      type: "DELETE_ROW"
    });
    expect(store.dispatch).toMatchSnapshot();

  });
});

const TestProvider = ({ store, children }) => (
  <Provider store={store}>{children}</Provider>
);

function testRender(ui, { store, ...otherOpts }) {
  return render(<TestProvider store={store}>{ui}</TestProvider>, otherOpts);
}

function makeTestStore() {
  const store = createStore(reducers, { table: testStore });

  store.dispatch = jest.fn();
  return store;
}

