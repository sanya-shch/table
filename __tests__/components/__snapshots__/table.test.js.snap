// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Table component table 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="tableCell "
        >
          615
        </td>
        <td
          class="tableCell "
        >
          676
        </td>
        <td
          class="tableCell "
        >
          335
        </td>
        <td>
          1626
        </td>
        <td
          class="delBtn"
        >
          Delete
        </td>
      </tr>
      <tr>
        <td
          class="tableCell "
        >
          802
        </td>
        <td
          class="tableCell "
        >
          703
        </td>
        <td
          class="tableCell "
        >
          254
        </td>
        <td>
          1759
        </td>
        <td
          class="delBtn"
        >
          Delete
        </td>
      </tr>
      <tr>
        <td
          class="tableCell "
        >
          982
        </td>
        <td
          class="tableCell "
        >
          919
        </td>
        <td
          class="tableCell "
        >
          748
        </td>
        <td>
          2649
        </td>
        <td
          class="delBtn"
        >
          Delete
        </td>
      </tr>
      <tr>
        <td>
          791.67
        </td>
        <td>
          602.33
        </td>
        <td>
          502
        </td>
        <td
          class="addBtn"
          data-testid="add"
        >
          Add
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`the Table component table 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "ADD_ROW",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`the Table component table 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "ADD_ROW",
      },
    ],
    Array [
      Object {
        "payload": "0",
        "type": "DELETE_ROW",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
